pkg/hlf/parser/parser_test.go

```diff
@@ -0,0 +1,76 @@
+package parser_test
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/anoideaopen/channel-transfer/pkg/hlf/parser"
+	"github.com/anoideaopen/glog"
+	"github.com/golang/protobuf/proto"
+	"github.com/hyperledger/fabric-protos-go/common"
+	"github.com/hyperledger/fabric-protos-go/peer"
+	"github.com/stretchr/testify/assert"
+)
+
+func readBlockFromFile(filePath string) (*common.Block, error) {
+	data, err := os.ReadFile(filePath)
+	if err != nil {
+		return nil, err
+	}
+
+	var block common.Block
+	if err := proto.Unmarshal(data, &block); err != nil {
+		return nil, err
+	}
+
+	return &block, nil
+}
+
+func TestExtractData_ExecuteTasksMethod(t *testing.T) {
+	blockFilePath := "./test-data/batcher-blocks/block_10"
```

Comment:
I don't like the presence of a binary file. Try to find another solution.

A binary file is an unknown that is not clear how to be maintained afterwards.

Maybe you should write an integration test instead of a unit test?